addon/usables/use-machine.js

```diff
@@ -69,15 +69,34 @@ export class MachineInterpreterManager {
     interpreter.setup();
   }
 
-  updateUsable() {
-    // not sure if it makes sense to support this - issue warning
-    warn(ARGS_STATE_CHANGE_WARNING, false, { id: 'statecharts.use-machine.args-state-change' });
+  updateUsable(bucket, configurableMachineDefinition) {
+    const { interpreter } = bucket;
+    const { args, _update } = configurableMachineDefinition;
+
+    if (_update) {
+      const { machine, context, config } = args;
+      _update({
+        machine,
+        context,
+        config,
```

Comment:
Hmm, I can't think of a use case where you'd need the new `machine` and `config` to decide if you want to restart to send an event 🤔 

But I guess that's up to the consume to use this responsibly.